api/V1/readme.md

```diff
@@ -0,0 +1,279 @@
+    - Reduces complexity of individual endpoints
+
+### 2. Protocol Independence
+A key design goal was to abstract away protocol-specific details:
+
+- **Removal of DAH Dependencies**: The API minimizes reliance on DataAvailabilityHeader (DAH) to:
+    - Reduce coupling to protocol internals
+    - Make the API more stable against protocol changes
+    - Simplify client implementations
+- **Data Root Proofs Focus**: Moving towards data root proofs instead of row roots because:
+    - Provides a more direct verification path
+    - Reduces complexity of proof verification
+    - Better aligns with future protocol optimizations
+- **Easier Migration**: Across all services, some requests and responses have V0/V1 suffixes. V0 supports row root-based proofs (matching our current implementation), while V1 uses data root verification. Both options are included to enable a smoother migration to the new API while data root proofs are still under development.
+
+### 3. Data Availability
+
+Data availability verification is fundamental to light node security and must be enforced rigorously through the API. The current implementation has two key limitations:
+1. No straightforward mechanism to verify data availability at a given height
+2. ShareAvailable API only verifies individual blocks, while proper DAS verification requires checking the entire sampling window preceding the requested height
+
+Proposed Solutions:
+
+1. **Enhanced ShareAvailable Verification**:
+   The ShareAvailable call will verify the continuous chain of blocks from the oldest block in the availability window up to the requested height. The call will return false if any block within this range is unavailable.
+
+2. **Integrated Availability Verification**:
+   A new `verify_available` option will be added to all data fetching methods (blobs, shares, etc). When enabled:
+  - The API will verify availability for both the requested block and all preceding blocks within the sampling window
+  - Data will only be returned if the entire chain segment is verified as available
+  - Failed availability checks will return a specific `data_not_available` error code
+
+```protobuf
+message Options {
+    // if verify_available is enabled, node will additionally ensure
+    // data availability on top of fetching requiested data
+    bool verify_available = 1;
+    // when proofs_only is enabled, proofs will be returned without sending data. 
+    bool proofs_only = 2;
+}
+```
+
+### 4. Error Response Structure
+```protobuf
+message Status {
+    StatusCode code = 1;
+    string message = 2;
+}
+```
+
+Advantages:
+- Consistent error reporting
+- Rich error information
+- Easy to extend
+- Supports both technical and user-friendly messages
+
+Some error codes already listed in the draft, however the list might be not full and would need to be revisited with each particular usecase in mind
+
+Standardized Status Codes
+```protobuf
+// Common status codes used across all services
+enum StatusCode {
```

Comment:
I do think we should have the possibility for a dev to know if an error stems from node or e.g. somewhere from app. Maybe that is already possible with the list somehow (e.g. not by the code but via the error string or sth?).